cmd/catalyst-node/catalyst-node.go

```diff
@@ -348,3 +353,34 @@ func main() {
 
 	os.Exit(exitCode)
 }
+
+func writeSerfConfig(config *agent.Config) (string, error) {
+	// Two steps to properly serialize this as JSON: https://github.com/spf13/viper/issues/816#issuecomment-1149732004
```

Comment:
Why do we need this hack/workaround? Can't we just create a type struct with `json` tags and marshal it?

Reply:
We totally could, there just already exists such a struct so I figured this would both save time and increase our confidence that we had used the correct JSON fields for Serf. A single-line ugly hack seems potentially worth it in that context?

Reply:
I'd probably favor having slightly more code, but no hacks ;) Anyway, up to you. Btw. you can check something similar I did here: https://github.com/livepeer/catalyst/pull/53/files#diff-a904e00073eb9c69508bd6a1c0ba249d3f7dc854094a8e6380db1bd9077c888cR9